test/client.test.js

```diff
@@ -3652,415 +3558,797 @@ describe('ACC Client', function () {
+
+              var ex = CampaignException.AEM_ASSET_UPLOAD_FAILED(
+                "The requested content does not exist",
+                400
+              );
+              await client.fileUploader
+                .uploadAemAsset(
+                  "https://author-stg.aem.adobe.com/adobe/repository/content/dam/projects/children-kids-group-eating-ice-cream-funny-party-72701973%20%281%29%20%2812%29%20%282%29%20%283%29.jpg;api=block_download"
+                )
+                .catch((e) => {
+                  expect(e).toMatchObject(ex);
+                });
 
-            // Call upload
-            const result = await client.fileUploader.upload({
-                type: 'text/html',
-                size: 12345
-            }, { action: "publishIfNeeded" });
+              client._transport.mockReturnValueOnce(Mock.LOGOFF_RESPONSE);
+              await client.NLWS.xtkSession.logoff();
+            });
 
-            expect(result).toMatchObject({
-                md5: "d8e8fca2dc0f896fd7cb4cb0031ba249",
-                name: "test.txt",
-                size: 12345,
-                type: "text/html",
-                url: "http://hello.com"
+            // Case 5: transport layer returns response as string instead of json when sdk
+            // used as package inside browser(Fetch API is used in browser while axios in node).
+            it("Should return correct response", async () => {
+              const client = await Mock.makeClient();
+              client._transport.mockReturnValueOnce(Mock.LOGON_RESPONSE);
+              await client.NLWS.xtkSession.logon();
+
+              const mockResponse = {
+                publishedURL:
+                  "http://trk-inst-xyz.camp.adobe.com/res/trk-inst/409afb8798180a36591456e152b6c406.jpeg",
+              };
+              const mockResponseString = JSON.stringify(mockResponse);
+              client._bearerToken = "Bearer 1234567890";
+              client._transport.mockReturnValueOnce(mockResponseString);
+              const response = await client.fileUploader.uploadAemAsset(
+                "https://author-stg.aem.adobe.com/adobe/repository/content/dam/projects/children-kids-group-eating-ice-cream-funny-party-72701973%20%281%29%20%2812%29%20%282%29%20%283%29.jpg;api=block_download"
+              );
+              expect(response).toMatchObject(mockResponse);
+
+              client._transport.mockReturnValueOnce(Mock.LOGOFF_RESPONSE);
+              await client.NLWS.xtkSession.logoff();
             });
+          });
         });
 
-        it("Should support 'className' action", async () => {
-            // Create a mock client and logon
+        describe("download", () => {
+          let fileOptions;
+          const fileExt = 'csv';
+          const fileMd5 = 'a88333b4e8b523a1c4fca8f3b378b8e0';
+
+          beforeEach(() => {
+            fileOptions = {
+              fileName: "myFile",
+              saveAsFile: false,
```

Comment:
What is saveAsFile?
It does not seem to be used, can you remove?

Reply:
Done